tests/test_object_manager.py

```diff
@@ -377,16 +396,35 @@ def test_bind_names_updates_host_record(self):
         ibom = om.InfobloxObjectManager(connector)
 
         ibom.bind_name_with_host_record(dns_view_name, ip, fqdn, extattrs)
+        connector.get_object.assert_called_once_with(
+            'record:host', {'view': dns_view_name, 'ipv4addr': ip},
+            extattrs=None, max_results=None,
+            force_proxy=mock.ANY, return_fields=mock.ANY)
+        connector.update_object.assert_called_once_with(
+            mock.ANY, {'name': fqdn}, mock.ANY)
 
-        matcher = PayloadMatcher({'view': dns_view_name,
-                                  PayloadMatcher.ANYKEY: ip})
+        connector.get_object.reset_mock()
+        connector.update_object.reset_mock()
```

Comment:
Similar comment like on L369, each time you need to do reset_mock() you probably need to create a new test case.
Then common code of these newly created test cases can be moved to helper methods.

As I see this test can be reworked into 3 tests and 1 helper method that wil contain all the common initialization and comparing logic.